### PR TITLE
React 16-alpha compatibility

### PR DESCRIPTION
**Pull request by andykog**
Resloves  #237, #233

This should be released as well: https://github.com/mobxjs/mobx-react-devtools/commit/198f753ea2bd9098a3dc044233facb53fdbf81a0

### CODE REVIEW

**Comment by andykog** [comment]
No idea why travis fails

File: src/inject.js [hunk 1]
```diff
@@ -88,6 +90,7 @@ function grabStoresByName(storeNames) {
  */
 export default function inject(/* fn(stores, nextProps) or ...storeNames */) {
   let grabStoresFn;
+  debugger;
```
Comment (mweststrate) [inline src/inject.js]: that one should be dropped ;-)

File: package.json [hunk 2]
```diff
@@ -28,30 +28,29 @@
   "homepage": "https://mobxjs.github.io/mobx",
   "peerDependencies": {
     "mobx": "^2.6.3 || ^3.0.0",
-    "react": "^0.13.0 || ^0.14.0 || ^15.0.0"
+    "react": "^0.13.0 || ^0.14.0 || ^15.0.0 || ^16.0.0-0"
```
Comment (mweststrate) [inline package.json]: What does the `-0` do? Making sure alpha is supported?

File: test/inject.js [hunk 3]
```diff
@@ -4,10 +4,6 @@ import { mount } from 'enzyme'
 import test from 'tape'
 import mobx, { action, observable, computed } from 'mobx'
 import { observer, inject, Provider } from '../'
-import $ from 'jquery'
```
Comment (mweststrate) [inline test/inject.js]: nice cleanup!

**Comment by mweststrate** [comment]
Great work @andykog!

File: package.json (hunk 2 above)
Comment (andykog) [inline package.json]: Alpha is supported thanks to that `-0` :-)

File: src/inject.js (hunk 1 above)
Comment (andykog) [inline src/inject.js]: Done (force-pushed)

File: package.json (hunk 2 above)
Comment (andykog) [inline package.json]: https://github.com/npm/node-semver/blob/master/semver.js#L74-L84

File: package.json [hunk 4]
```diff
@@ -28,30 +28,29 @@
...
+    "react": "^15.5.3",
+    "react-addons-test-utils": "^15.5.1",
+    "react-dom": "^15.5.3",
     "tap-spec": "^4.1.1",
-    "tape": "^4.2.2",
-    "tape-run": "2.1.0",
-    "typescript": "^2.1.4",
-    "webpack": "^1.13.1"
+    "tape": "^4.6.3",
+    "tape-run": "3.0.0",
```
Comment (mweststrate) [inline package.json]: Ok, this commit seems to fix the travis issue. @andykog  can you downgrade `tape-run` in this branch?

**Comment by mweststrate** [comment]
Awesome stuff @andykog. Thanks!

**Comment by andykog** [comment]
@mweststrate, you was to rapid merging this, I've actually reverted some changes to make travis work, pushed https://github.com/mobxjs/mobx-react/commit/4f8a8603af24a921a93d16eef04ca2f323f0b46a and released as `mobx-react@4.1.7`

**Pull request merged**